internal/pkg/agent/application/filelock/app_locker.go

```diff
@@ -8,41 +8,28 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
-
-	"github.com/gofrs/flock"
 )
 
 // ErrAppAlreadyRunning error returned when another elastic-agent is already holding the lock.
 var ErrAppAlreadyRunning = fmt.Errorf("another elastic-agent is already running")
 
 // AppLocker locks the agent.lock file inside the provided directory.
 type AppLocker struct {
-	lock *flock.Flock
+	*FileLocker
 }
 
 // NewAppLocker creates an AppLocker that locks the agent.lock file inside the provided directory.
 func NewAppLocker(dir, lockFileName string) *AppLocker {
 	if _, err := os.Stat(dir); os.IsNotExist(err) {
 		_ = os.Mkdir(dir, 0755)
 	}
-	return &AppLocker{
-		lock: flock.New(filepath.Join(dir, lockFileName)),
-	}
+
+	lock, _ := NewFileLocker(filepath.Join(dir, lockFileName), WithCustomNotLockedError(ErrAppAlreadyRunning))
```

Comment:
we don't check error or nil, we can panic later

Reply:
Creation of a FileLocker is never blocking (it doesn't try to lock at creation). Moreover, AppLocker was non-blocking in its previous version (using only `flock.TryLock()`) and the impl with FileLocker maintains that behavior.

As for the error handling, our options here are limited since we have no errors to return or logger to write out the error, I will add a panic() which should never happen (the only case where we can get an error at FileLocker creation is by specifiying a zero timeout, which AppLocker doesn't do)